JSON.md

```diff
@@ -172,3 +172,388 @@ cfbs init && cfbs add https://github.com/cfengine/some-repo
   ]
 }
 ```
+
+
+## Modules with input
+Some modules allow for users to add module input by responding to questions
+expressed under the `"input"` attribute in `cfbs.json`. User input can be added
+using the `cfbs input <module-name>` command, which stores responses in
+`./<module-name>/input.json`. These responses are translated into augments which
+will be added to `./out/masterfiles/def.json` during `cfbs build`.
+
+### Create single file example:
+The `"input"` attribute takes a list of input definitions as illustrated below.
+
+```json
+{
+  "name": "Example",
+  "type": "policy-set",
+  "description": "Example description",
+  "build": [
+    {
+      "name": "create-single-file",
+      "description": "Create a single file.",
+      "url": "https://github.com/cfengine/example-module.git",
+      "commit": "d95774c8c59a2895c677624851ef4ad9d5e0d02d",
+      "dependencies": ["autorun"],
+      "added_by": "cfbs add",
+      "steps": [
+        "copy ./create-single-file.cf services/autorun/create-single-file.cf"
+      ],
+      "input": [
+        {
+          "type": "string",
+          "variable": "filename",
+          "label": "Filename",
+          "question": "What file should this module create?"
+        }
+      ]
+    }
+  ]
+}
+```
+
+From the example above, we can see that the `"input"` list contains one input
+definition. By running the command `cfbs input create-single-file`, the input
+definition will be copied into `./create-single-file/input.json` along with the
+user responses.
+
+```
+$ cfbs input create-single-file
+Adding input for module 'create-single-file':
+What file should this module create? /tmp/create-single-file.txt
+$ cat ./create-single-file/input.json
+[
+  {
+    "type": "string",
+    "variable": "filename",
+    "label": "Filename",
+    "question": "What file should this module create?",
+    "response": "/tmp/create-single-file.txt"
+  }
+]
+```
+
+By running `cfbs build`, augments will be generated from
+`./create-single-file/input.json` and added to `./out/masterfiles/def.json`
+
+```
+$ cfbs build
+--snip--
+Build complete, ready to deploy 🐿
+ -> Directory: out/masterfiles
+ -> Tarball:   out/masterfiles.tgz
+To install on this machine: sudo cfbs install
+To deploy on remote hub(s): cf-remote deploy
+$ cat ./out/masterfiles/def.json
+{
+  "classes": {
+    "services_autorun": ["any"]
+  },
+  "variables": {
+    "cfbs:create_single_file.filename": {
+      "value": "/tmp/create-single-file.txt",
+      "comment": "Added by 'cfbs input'"
+    }
+  }
+}
+```
+
```

Comment:
Show that a buildstep is added in `cfbs.json` after build to make this more clear
```json
"steps": [
        "copy ./create-multiple-files-with-content.cf services/autorun/create-multiple-files-with-content.cf",
        "input ./input.json def.json"
      ],
```